components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java

```diff
@@ -1425,17 +1428,32 @@ private Cookie getTokenIdCookie(HttpServletRequest req) {
      * @param sessionId
      * @param req
      * @param resp
+     * @param loginTenantDomain
      */
     private void storeTokenIdCookie(String sessionId, HttpServletRequest req, HttpServletResponse resp,
-                                    String tenantDomain) {
+                                    String tenantDomain, String loginTenantDomain) {
+
         ServletCookie samlssoTokenIdCookie = new ServletCookie(SAML_SSO_TOKEN_ID_COOKIE, sessionId);
         IdentityCookieConfig samlssoTokenIdCookieConfig = IdentityUtil
                 .getIdentityCookieConfig(SAML_SSO_TOKEN_ID_COOKIE);
         int defaultMaxAge = IdPManagementUtil.getIdleSessionTimeOut(tenantDomain);
 
         samlssoTokenIdCookie.setSecure(true);
         samlssoTokenIdCookie.setHttpOnly(true);
-        samlssoTokenIdCookie.setPath("/");
+
+        boolean isTenantQualifiedCookie = false;
+        if (IdentityTenantUtil.isTenantedSessionsEnabled() &&
+                sessionId.endsWith(SAMLSSOConstants.TENANT_QUALIFIED_TOKEN_ID_COOKIE_SUFFIX)) {
```

Comment:
What if sessionId has null value?

Reply:
There is no possibility of having null value for the sessionId. 
In here we check the sessionId value and it we always pass a not null value to this method.
https://github.com/wso2-extensions/identity-inbound-auth-saml/blob/719717cbf2d0ac4d4ba8bee0d307d1aabcc1224f/components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/servlet/SAMLSSOProviderServlet.java#L1172-L1174